src/broker/overlay.c

```diff
@@ -516,6 +538,8 @@ static void overlay_mcast_child (struct overlay *ov, const flux_msg_t *msg)
             if (overlay_mcast_child_one (ov, msg, child) < 0) {
                 if (errno == EHOSTUNREACH) {
                     child->connected = false;
+                    overlay_child_log (ov->h, child, "Mcast failed: %s",
+                                       flux_strerror (errno));
```

Comment:
In the testsuite, for some reason `flux_strerror (errno)` appears to return an empty string here (at least on my system). I have no idea how this could be the case:
```
	flux dmesg|grep broker.debug|grep "overlay child"

2021-06-09T14:38:40.594423Z broker.debug[0]: overlay child 1 62e99b77-5f30-4d67-abb0-06f51e063380 connnected=true Sent hello
2021-06-09T14:38:40.711337Z broker.debug[0]: overlay child 2 01c09cde-bf7f-43d2-b603-45849e4a32f4 connnected=true Sent hello
2021-06-09T14:39:00.358933Z broker.debug[0]: overlay child 2 01c09cde-bf7f-43d2-b603-45849e4a32f4 connnected=false Sent DISCONNECT
2021-06-09T14:39:03.221390Z broker.debug[0]: overlay child 2 3712f6fb-8eb2-416e-b732-6fde0d0e60a9 connnected=true Sent hello
2021-06-09T14:39:05.229186Z broker.debug[0]: overlay child 2 3712f6fb-8eb2-416e-b732-6fde0d0e60a9 connnected=false Mcast failed: 
ok 18 - dump overlay child logs from leader broker
```

Since this log message is only emitted when `errno == EHOSTUNREACH`, perhaps a custom error message could be used here instead?

Edit: this error is only emitted when running the testcase under `FLUX_TEST_VALGRIND=t`. Maybe that is part of the problem.

Reply:
Well that's unexpected!  I went ahead and made the change you suggested.

I thought I had reviewed strerror code and determined that it was safe to use the non-thread-safe version in flux, but of course we do have multiple threads potentially going at it in the broker.  I'll open an issue to revisit that.